src/schema/schema-generator.ts

```diff
@@ -604,4 +638,104 @@ export abstract class SchemaGenerator {
       return this.objectTypesInfo.find(objectType => objectType.target === resolvedType)!.type;
     };
   }
+
+  private static createClassDirectiveNodes(
+    name: string,
+    directiveMetas?: DirectiveMetadata[],
+    kind: ObjectTypeDefinitionNode["kind"] = "ObjectTypeDefinition",
+  ): TypeDefinitionNode | undefined {
+    if (!directiveMetas || !directiveMetas.length) {
+      return;
+    }
+
+    return {
+      kind,
+      name: {
+        kind: "Name",
+        value: name,
+      },
+      interfaces: [],
+      directives: directiveMetas
+        .reverse()
+        .map(this.createDirectiveNodes)
+        .reduce((acc, val) => acc.concat(val), []),
+    };
+  }
+
+  private static createFieldDirectiveNodes(
+    name: string,
+    directiveMetas?: DirectiveMetadata[],
+    kind: FieldDefinitionNode["kind"] | InputValueDefinitionNode["kind"] = "FieldDefinition",
+  ): FieldDefinitionNode | InputValueDefinitionNode | undefined {
+    if (!directiveMetas || !directiveMetas.length) {
+      return;
+    }
+
+    return {
+      kind,
+      type: {
+        kind: "NamedType",
+        name: {
+          kind: "Name",
+          value: name,
+        },
+      },
+      name: {
+        kind: "Name",
+        value: name,
+      },
+      directives: directiveMetas
+        .reverse()
+        .map(this.createDirectiveNodes)
+        .reduce((acc, val) => acc.concat(val), []),
+    };
+  }
+
+  private static createDirectiveNodes(directive: DirectiveMetadata): DirectiveNode[] {
+    const { nameOrSDL, args } = directive;
+
+    if (!nameOrSDL.startsWith("@")) {
+      return [
+        {
+          kind: "Directive",
+          name: {
+            kind: "Name",
+            value: nameOrSDL,
+          },
+          arguments: Object.keys(args).map(arg => ({
+            kind: "Argument",
+            name: {
+              kind: "Name",
+              value: arg,
+            },
+            value: parseValue(args[arg]),
+          })),
+        },
+      ];
+    }
+
+    let directives: DirectiveNode[] = [];
+
+    try {
+      const parsed = parse(`type String ${nameOrSDL}`);
+
+      const definitions = parsed.definitions as ObjectTypeDefinitionNode[];
+
+      if (definitions && definitions.length > 0) {
+        definitions.forEach(def => {
+          if (def.directives && def.directives.length > 0) {
+            directives = [...directives, ...def.directives];
+          }
+        });
+      }
+    } catch (err) {
+      /** noop */
```

Comment:
why noop? we should throw error so the user can fix his directive sdl

Reply:
I'm throwing an error below if there are no directives, but yes, I can create a specific parsing error.